tests/MongoDB.Driver.Core.Tests/Specifications/connection-monitoring-and-pooling/ConnectionMonitoringAndPoolingTestRunner.cs

```diff
@@ -0,0 +1,613 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers;
+using MongoDB.Bson.TestHelpers.JsonDrivenTests;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Clusters;
+using MongoDB.Driver.Core.Configuration;
+using MongoDB.Driver.Core.ConnectionPools;
+using MongoDB.Driver.Core.Connections;
+using MongoDB.Driver.Core.Events;
+using MongoDB.Driver.Core.Helpers;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.Servers;
+using Moq;
+using Xunit;
+
+namespace MongoDB.Driver.Specifications.connection_monitoring_and_pooling
+{
+    public class ConnectionMonitoringAndPoolingTestRunner
+    {
+        #region static
+        private static readonly string[] __alwaysIgnoredEvents =
+        {
+            nameof(ConnectionPoolOpeningEvent),
+            nameof(ConnectionPoolAddingConnectionEvent),
+            nameof(ConnectionPoolAddedConnectionEvent),
+            nameof(ConnectionPoolCheckingInConnectionEvent),
+            nameof(ConnectionPoolRemovingConnectionEvent),
+            nameof(ConnectionPoolRemovedConnectionEvent),
+            nameof(ConnectionPoolClosingEvent),
+            nameof(ConnectionPoolClearingEvent),
+            nameof(ConnectionOpeningEvent),
+            nameof(ConnectionClosingEvent)
+        };
+        #endregion
+
+        [SkippableTheory]
+        [ClassData(typeof(TestCaseFactory))]
+        public void RunTestDefinition(JsonDrivenTestCase testCase)
+        {
+            var connectionMap = new ConcurrentDictionary<string, IConnection>();
+            var eventCapturer = new Lazy<EventCapturer>(isThreadSafe: true);
+            var tasks = new ConcurrentDictionary<string, Task>();
+
+            var test = testCase.Test;
+            JsonDrivenHelper.EnsureAllFieldsAreValid(test, "_path", "version", "style", "description", "poolOptions", "operations", "error", "events", "ignore", "async");
+            EnsureAvailableStyle(test);
+            ResetConnectionId();
+
+            var connectionPool = new Lazy<IConnectionPool>(
```

Comment:
I can't think of any reason this needs to be `Lazy`.

I removed the `Lazy` and tests still pass.

Am I missing something?

Reply:
The same as above